velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -715,69 +715,90 @@ TypePtr ReaderBase::convertType(
           schemaElement.__isset.type_length,
       "FIXED_LEN_BYTE_ARRAY requires length to be set");
 
+  static std::string_view kTypeMappingErrorFmtStr =
+      "Converted type {} is not allowed for requested type {}";
   if (schemaElement.__isset.converted_type) {
     switch (schemaElement.converted_type) {
       case thrift::ConvertedType::INT_8:
-        VELOX_CHECK_EQ(
-            schemaElement.type,
-            thrift::Type::INT32,
-            "INT8 converted type can only be set for value of thrift::Type::INT32");
-        return TINYINT();
-
-      case thrift::ConvertedType::INT_16:
-        VELOX_CHECK_EQ(
-            schemaElement.type,
-            thrift::Type::INT32,
-            "INT16 converted type can only be set for value of thrift::Type::INT32");
-        return SMALLINT();
-
-      case thrift::ConvertedType::INT_32:
-        VELOX_CHECK_EQ(
-            schemaElement.type,
-            thrift::Type::INT32,
-            "INT32 converted type can only be set for value of thrift::Type::INT32");
-        return INTEGER();
-
-      case thrift::ConvertedType::INT_64:
-        VELOX_CHECK_EQ(
-            schemaElement.type,
-            thrift::Type::INT64,
-            "INT64 converted type can only be set for value of thrift::Type::INT64");
-        return BIGINT();
-
       case thrift::ConvertedType::UINT_8:
         VELOX_CHECK_EQ(
             schemaElement.type,
             thrift::Type::INT32,
-            "UINT_8 converted type can only be set for value of thrift::Type::INT32");
+            "{} converted type can only be set for value of thrift::Type::INT32",
+            schemaElement.converted_type == thrift::ConvertedType::INT_8
```

Comment:
Simply `schemaElement.converted_type` should do here. I see that there is a fmt formatter for `facebook::velox::parquet::thrift::ConvertedType::type` inside ParquetThriftTypes.h. Or you can use `to_string(schemaElement.converted_type)`